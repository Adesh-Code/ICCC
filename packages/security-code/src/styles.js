const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
