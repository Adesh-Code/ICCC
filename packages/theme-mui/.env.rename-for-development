# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE