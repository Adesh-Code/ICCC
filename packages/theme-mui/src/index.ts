// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
