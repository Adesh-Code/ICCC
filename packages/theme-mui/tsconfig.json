{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
