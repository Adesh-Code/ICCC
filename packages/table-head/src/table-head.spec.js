import React from "react";
import TableHead from "./table-head";
import { render, screen } from "@testing-library/react";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = ({
  headCells = [
    { id: "name", label: "Dessert (100g serving)", align: "right" },
    { id: "calories", label: "Calories" },
    { id: "fat", label: "Fat (g)" },
    { id: "carbs", label: "Carbs (g)" },
    { id: "protein", label: "Protein (g)" }
  ],
  rowCount = 3,
  ...otherProps
} = {}) => {
  return render(
    <ThemeProvider theme={light}>
      <table>
        <TableHead headCells={headCells} rowCount={rowCount} {...otherProps} />
      </table>
    </ThemeProvider>
  );
};

describe("Table head", () => {
  it("renders", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders with checkbox", () => {
    arrange({ enableSelect: true });
    expect(screen.getByRole("checkbox")).toBeInTheDocument();
  });

  it("renders without enable select class name", () => {
    arrange();
    const firstCell = screen.getAllByRole("columnheader")[0];
    expect(firstCell).not.toHaveClass("DhigTableHead--cell--enableSelect");
  });

  it("renders with enable select class name", () => {
    arrange({ enableSelect: true });
    const firstCell = screen.getAllByRole("columnheader")[0];
    expect(firstCell).toHaveClass("DhigTableHead--cell--enableSelect");
  });

  it("renders indeterminate checkbox when not all rows are selected", () => {
    arrange({
      enableSelect: true,
      selectedRowCount: 1
    });
    const checkbox = screen.getByRole("checkbox");
    const checkboxIcon = checkbox.parentElement.lastChild.firstChild.firstChild;
    expect(checkboxIcon).toHaveAttribute("id", "indeterminate");
  });

  it("renders with sort label", () => {
    arrange({
      enableSort: true,
      onRequestSort: () => {},
      order: "asc",
      orderBy: "name"
    });
    const sortButton = screen.getAllByRole("button")[0];
    expect(sortButton).toHaveClass("MuiTableSortLabel-root");
  });

  it("renders with sort class name", () => {
    arrange({
      enableSort: true,
      onRequestSort: () => {},
      order: "asc",
      orderBy: "name"
    });
    const secondCell = screen.getAllByRole("columnheader")[1];
    expect(secondCell).toHaveClass("DhigTableHead--cell--sort");
  });

  it("handles alignment", () => {
    arrange();
    const row = screen.getByRole("row");
    expect(row.firstChild).toHaveClass("MuiTableCell-alignRight");
    expect(row.lastChild).toHaveClass("MuiTableCell-alignLeft");
  });
});
