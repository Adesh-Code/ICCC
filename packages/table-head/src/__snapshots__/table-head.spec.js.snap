// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table head renders 1`] = `
<div>
  <table>
    <thead
      class="MuiTableHead-root emotion-0"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head emotion-1"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium DhigTableHead--cell emotion-2"
          scope="col"
          tabindex="0"
        >
          Dessert (100g serving)
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium DhigTableHead--cell emotion-3"
          scope="col"
          tabindex="0"
        >
          Calories
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium DhigTableHead--cell emotion-3"
          scope="col"
          tabindex="0"
        >
          Fat (g)
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium DhigTableHead--cell emotion-3"
          scope="col"
          tabindex="0"
        >
          Carbs (g)
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium DhigTableHead--cell emotion-3"
          scope="col"
          tabindex="0"
        >
          Protein (g)
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
